src/main/java/core/controller/AuthenticationController.java

```diff
@@ -0,0 +1,27 @@
+package core.controller;
+
+import core.model.dto.UserRequestDto;
+import core.security.AuthenticationService;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class AuthenticationController {
+    private final AuthenticationService authenticationService;
+    
+    public AuthenticationController(AuthenticationService authenticationService) {
+        this.authenticationService = authenticationService;
+    }
+    
+    @PostMapping("/register")
+    public String registerUser(@RequestBody UserRequestDto requestDto) {
+        try {
+            authenticationService.register(requestDto.getEmail(), requestDto.getPassword());
+        } catch (Exception e) {
+            return "The user with specified email already exists. "
+                   + "Choose another email and try again!";
+        }
+        return "User " + requestDto.getEmail() + " was successfully registered";
```

Comment:
Nope. Better return any DTO object or leave it void. If you will have a `void` method then in case if everything is ok the user will receive HTTP status 200. In case of error, the user will receive a 4xx or 5xx status code.

Reply:
Ok, let it be void